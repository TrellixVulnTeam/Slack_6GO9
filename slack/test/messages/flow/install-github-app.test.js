const { InstallGitHubApp } = require('../../../lib/messages/flow');

describe('InstallGitHubApp message rendering', () => {
  test('works without a note', async () => {
    const message = new InstallGitHubApp({
      installLink: 'https://github.com/apps/url-to-slack-app',
      owner: {
        id: 5,
      },
      gitHubUser: {
        id: '5',
      },
    });
    expect(message.toJSON()).toMatchSnapshot();
  });
  test('works with if owner is user ', async () => {
    const message = new InstallGitHubApp({
      installLink: 'https://github.com/apps/url-to-slack-app',
      owner: {
        id: 3,
        type: 'User',
      },
      gitHubUser: {
        id: '5',
      },
    });
    expect(message.toJSON()).toMatchSnapshot();
  });
  test('works if owner is organization', async () => {
    const message = new InstallGitHubApp({
      installLink: 'https://github.com/apps/url-to-slack-app',
      owner: {
        id: 3,
        type: 'Organization',
      },
      gitHubUser: {
        id: '5',
      },
    });
    expect(message.toJSON()).toMatchSnapshot();
  });
});
