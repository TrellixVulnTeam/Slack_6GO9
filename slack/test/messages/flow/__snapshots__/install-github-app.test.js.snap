// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallGitHubApp message rendering works if owner is organization 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "style": "primary",
          "text": "Install GitHub App",
          "type": "button",
          "url": "https://github.com/apps/url-to-slack-app",
        },
      ],
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Either the app isn't installed on your repository or the repository does not exist. Install it to proceed.
_Note: You need to be an organization owner to install the app (or ask one to install it for you)._",
    },
  ],
  "response_type": "ephemeral",
}
`;

exports[`InstallGitHubApp message rendering works with if owner is user  1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "style": "primary",
          "text": "Install GitHub App",
          "type": "button",
          "url": "https://github.com/apps/url-to-slack-app",
        },
      ],
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Either the app isn't installed on your repository or the repository does not exist. Install it to proceed.
_Note: You will need to ask the owner of the repository to install it for you. Give them <https://github.com/apps/url-to-slack-app|this link.>_",
    },
  ],
  "response_type": "ephemeral",
}
`;

exports[`InstallGitHubApp message rendering works without a note 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "style": "primary",
          "text": "Install GitHub App",
          "type": "button",
          "url": "https://github.com/apps/url-to-slack-app",
        },
      ],
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Either the app isn't installed on your repository or the repository does not exist. Install it to proceed.",
    },
  ],
  "response_type": "ephemeral",
}
`;
